tests/system/src/test/java/test/com/sun/javafx/tk/quantum/SystemMenuBarTest.java

```diff
@@ -0,0 +1,351 @@
+        menuBar.getMenus().add(menu);
+        menuBar.setUseSystemMenuBar(true);
+        root.getChildren().add(menuBar);
+
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
+        stage.show();
+        final ArrayList<WeakReference<MenuBase>> uncollectedMenus = new ArrayList<>();
+        GlassSystemMenuShim gsmh = new GlassSystemMenuShim();
+        assumeTrue("SystemMenu only supported on MacOS", gsmh.isSupported());
+        Menu m1 = new Menu("Menu");
+
+        MenuBase menuBase = GlobalMenuAdapter.adapt(m1);
+        ArrayList<MenuBase> menus = new ArrayList<>();
+        gsmh.createMenuBar();
+        for (int i = 0; i < 100; i++) {
+            Platform.runLater(() -> {
+                gsmh.setMenus(List.of(menuBase));
+            });
+        }
+        Platform.runLater(() -> {
+            int strongCount = 0;
+            final List<WeakReference<com.sun.glass.ui.Menu>> u2 = gsmh.getWeakMenuReferences();
+            for (WeakReference<com.sun.glass.ui.Menu> wr : u2) {
+                if (!JMemoryBuddy.checkCollectable(wr)) {
+                    strongCount++;
+                    assertTrue("Too many references", strongCount < 2);
+                }
+            }
+            assertEquals(1, strongCount, "Exactly one reference should be reachable");
+            memoryFocusLatch.countDown();
+        });
+    }
+
+    CountDownLatch removeMenuLatch = new CountDownLatch(1);
+
+    @Test
+    public void testRemoveMenu() throws InterruptedException {
+        failed.set(false);
+        Util.runAndWait(() -> {
+            Thread.currentThread().setUncaughtExceptionHandler((t,e) -> {
+                e.printStackTrace();
+                failed.set(true);
+                removeMenuLatch.countDown();
+            });
+            createRemoveMenuStage();
+        });
+        removeMenuLatch.await();
+        assertFalse(failed.get());
+    }
+
+    public void createRemoveMenuStage() {
+        Stage stage = new Stage();
+        VBox root = new VBox();
+
+        final MenuBar menuBar = new MenuBar();
+
+        Menu mainMenu = new Menu("MainMenu");
+        Menu subMenu = new Menu("SubMenu");
+        subMenu.getItems().addAll(new MenuItem("submenuitem1"), new MenuItem("submenuitem2"));
+        mainMenu.getItems().add(subMenu);
+        menuBar.getMenus().add(mainMenu);
+
+        menuBar.setUseSystemMenuBar(true);
+        root.getChildren().add(menuBar);
+
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
+        stage.show();
+
+        Platform.runLater(() -> {
+            mainMenu.getItems().clear();
+            mainMenu.getItems().add(subMenu);
+            subMenu.getItems().addAll(new MenuItem("new item 1"), new MenuItem("new item 2"));
+            removeMenuLatch.countDown();
+        });
+    }
+
+    @Test
+    public void testJDK8309935() {
+        MenuBar menuBar = new MenuBar();
+        AtomicReference<Throwable> throwableRef = new AtomicReference();
```

Comment:
Add missing `<>` to avoid raw use of `AtomicReference`

Reply:
done